packages/block-editor/src/components/block-list/block.js

```diff
@@ -578,7 +583,7 @@ function BlockListBlockProvider( props ) {
 				mayDisplayControls:
 					_isSelected ||
 					( isFirstMultiSelectedBlock( clientId ) &&
-						getMultiSelectedBlockClientIds().every(
+						getSelectedBlockClientIdsUnmemoized().every(
```

Comment:
Maybe `mayDisplayControls` etc. should be selectors instead, but it's private anyway so it matters little.